Specs/Core/OrientedBoundingBoxSpec.js

```diff
@@ -0,0 +1,417 @@
+        var rect = new BoundingRectangle(1.0, 2.0, 3.0, 4.0);
+        var angle = CesiumMath.PI_OVER_TWO;
+        var box = OrientedBoundingBox.fromBoundingRectangle(rect, angle);
+        var rotation = Matrix3.fromRotationZ(angle);
+        expect(box.center).toEqual(new Cartesian3(-1.0, 3.5, 0.0));
+
+        var rotScale = new Matrix3();
+        Matrix3.multiply(rotation, Matrix3.fromScale(new Cartesian3(1.5, 2.0, 0.0)), rotScale);
+        expect(box.halfAxes).toEqualEpsilon(rotScale, CesiumMath.EPSILON15);
+    });
+
+    it('fromBoundingRectangle creates an OrientedBoundingBox with a result parameter', function() {
+        var rect = new BoundingRectangle(1.0, 2.0, 3.0, 4.0);
+        var angle = CesiumMath.PI_OVER_TWO;
+        var result = new OrientedBoundingBox();
+        var box = OrientedBoundingBox.fromBoundingRectangle(rect, angle, result);
+        expect(box).toBe(result);
+        var rotation = Matrix3.fromRotationZ(angle);
+
+        expect(box.center).toEqual(new Cartesian3(-1.0, 3.5, 0.0));
+
+        var rotScale = new Matrix3();
+        Matrix3.multiply(rotation, Matrix3.fromScale(new Cartesian3(1.5, 2.0, 0.0)), rotScale);
+        expect(box.halfAxes).toEqualEpsilon(rotScale, CesiumMath.EPSILON15);
+    });
+
+    /**
+     * @param {Cartesian3} center
+     * @param {Matrix3} halfAxes
+     */
+    var planeIntersectTestCornersEdgesFaces = function(center, axes) {
+        var SQRT1_2 = Math.pow(1.0 / 2.0, 1 / 2.0);
+        var SQRT1_3 = Math.pow(1.0 / 3.0, 1 / 2.0);
+        var SQRT3_4 = Math.pow(3.0 / 4.0, 1 / 2.0);
+
+        var box = new OrientedBoundingBox(center, Matrix3.multiplyByScalar(axes, 0.5, new Matrix3()));
+
+        var planeNormXform = function(nx, ny, nz, dist) {
+            var n = new Cartesian3(nx, ny, nz);
+            var arb = new Cartesian3(357, 924, 258);
+            var p0 = Cartesian3.normalize(n, new Cartesian3());
+            Cartesian3.multiplyByScalar(p0, -dist, p0);
+            var tang = Cartesian3.cross(n, arb, new Cartesian3());
+            Cartesian3.normalize(tang, tang);
+            var binorm = Cartesian3.cross(n, tang, new Cartesian3());
+            Cartesian3.normalize(binorm, binorm);
+
+            Matrix3.multiplyByVector(axes, p0, p0);
+            Matrix3.multiplyByVector(axes, tang, tang);
+            Matrix3.multiplyByVector(axes, binorm, binorm);
+            Cartesian3.cross(tang, binorm, n);
+            Cartesian3.normalize(n, n);
+
+            Cartesian3.add(p0, center, p0);
+            var d = -Cartesian3.dot(p0, n);
+            if (Math.abs(d) > 0.0001 && Cartesian3.magnitudeSquared(n) > 0.0001) {
+                return new Cartesian4(n.x, n.y, n.z, d);
+            } else {
+                return undefined;
+            }
+        };
+
+        var pl;
+
+
+        // Tests against faces
+
+        pl = planeNormXform(+1.0, +0.0, +0.0,          + 0.50001); if (pl) { expect(box.planeIntersect(pl)).toEqual(Intersect.INSIDE); }
```

Comment:
I just had it like this since I was doing a lot of block editing across all of these lines. It should be removed now that that's done?